topy/topy.py

```diff
@@ -121,9 +121,22 @@ def sanitize_filename(filename):
 def print_diff(filename, old, new, stream=sys.stdout):
     """Diffs the `old` and `new` strings and prints as unified diff to file-like object `stream`."""
 
-    # TODO: color output for terminals
     lines = unified_diff(old.splitlines(True), new.splitlines(True), filename, filename)
-    stream.writelines(lines)
+    lines_color = add_output_color(lines)
+    stream.writelines(lines_color if not None else lines)  # print in colour if available
+
+
+def add_output_color(lines):
+    """Adds color to the output of the diff lines."""
+    try:
+        lines_list = list(lines)
+        for index, line in enumerate(lines_list):
+            lines_list[index] = f'\033[1;32m{line}\033[0;32m' if line.startswith('+') else (
+                f'\033[1;31m{line}\033[0;31m' if line.startswith('-') else f'\033[0m{line}')
+        return lines_list
+    except TypeError:
+        log.info("Could not convert %s to a list", lines)
+        return None
```

Comment:
I think I was trying to protect against the input parameter not being a sequence, but since unified_diff() returns an iterator, like you said, I can probably remove this exception